numpy/core/src/multiarray/compiled_base.c

```diff
@@ -415,6 +417,22 @@ arr_insert(PyObject *NPY_UNUSED(self), PyObject *args, PyObject *kwdict)
         goto fail;
     }
     numvals = PyArray_SIZE(avals);
+
+    if (!numvals) {
+        /* nothing to insert! fail unless none of mask is true */
+        PyObject *const any = PyArray_Any(amask, NPY_MAXDIMS, NULL);
```

Comment:
Interesting approach... Since `amask` is a contiguous array of `npy_intp` type, wouldn't it make more sense to traverse it with a loop directly? Something like:

```
npy_intp amask_size = PyArray_SIZE(amask);
npy_intp *data = PyArray_DATA(amask);
int fail = 0;

while (amask_size-- && !fail) {
    fail = (*data++ != 0);
}
```